test/integration/xfcc_integration_test.cc

```diff
@@ -238,5 +238,709 @@ TEST_P(XfccIntegrationTest, NonTlsEnforceSanitize) {
   startTestServerWithXfccConfig("forward_only", "");
   testRequestAndResponseWithXfccHeader(makeClientConnection(), previous_xfcc_, "");
 }
+
+TEST_P(XfccIntegrationTest, TagExtractedNameGenerationTest) {
+  // Note: the test below is meant to check that default tags are being extracted correctly with
+  // real-ish input stats. If new stats are added, this test will not break because names that do
+  // not exist in the map are not checked. However, if stats are modified the below maps should be
+  // updated (or regenerated by printing in map literal format). See commented code below to
+  // regenerate the maps. Note: different maps are needed for ipv4 and ipv6, so when regenerating,
+  // the printout needs to be copied from each test parameterization and pasted into the respective
+  // case in the switch statement below.
+
+  startTestServerWithXfccConfig("forward_only", "");
+
+  // Commented sample code to regenerate the map literals used below in the test log if necessary:
+
+  // std::cout << "tag_extracted_counter_map = {";
+  // std::list<Stats::CounterSharedPtr> counters = test_server_->counters();
+  // for (auto it = counters.begin(); it != counters.end(); ++it) {
+  //   if (it != counters.begin()) {
+  //     std::cout << ",";
+  //   }
+  //   std::cout << std::endl << "{\"" << (*it)->name() << "\", \"" << (*it)->tagExtractedName() <<
+  //   "\"}";
+  // }
+  // std::cout << "};" << std::endl;
+  // std::cout << "tag_extracted_gauge_map = {";
+  // std::list<Stats::GaugeSharedPtr> gauges = test_server_->gauges();
+  // for (auto it = gauges.begin(); it != gauges.end(); ++it) {
+  //   if (it != gauges.begin()) {
+  //     std::cout << ",";
+  //   }
+  //   std::cout << std::endl << "{\"" << (*it)->name() << "\", \"" << (*it)->tagExtractedName() <<
+  //   "\"}";
+  // }
+  // std::cout << "};" << std::endl;
```

Comment:
This is fine with me if OK with @mattklein123. Might also be a valid thing to do to factor this code into a function that is used to both generate the golden file and the current state for comparison, but I think you've got most of the win here by just having some real state being validated against your regexes.